### ISSUE TITLE
Doubts in the reproduction process

### ISSUE THREAD

**Issue by baigeiguai** [Ali-E/AMUN#1]
Dear author, thank you for your contributions. I tried to reproduce the results in your work, but when I tried to reproduce the results in Table 1 of the paper, we re-ran the AMUN method with RANDOM FORGET=10%, but we got an UNLEARN ACC result of 99.88 and the model seemed to be not unlearn at all. We are very surprised by this phenomenon and look forward to your answer to our doubts.
I ran the following commands in sequence during the reproduction.

```
python batch_job_submit.py --job_type train --dataset cifar --model ResNet18 --seed 3407 --device=7

python batch_job_submit.py --job_type advdata --method orig --seed 3407 --dataset cifar --model_path  ./logs/scratch/cifar_ResNet18_unnorm/vanilla_orig_wBN_3407/  --epoch 130 

python forget_index_generate.py cifar 50000 5000

python batch_job_submit.py --job_type advunlearn --seed 3407    --dataset cifar --adv_images logs/scratch/cifar_ResNet18_unnorm/vanilla_orig_wBN_3407/adv_data/seed_3407/adv_tensor.pt  --adv_delta logs/scratch/cifar_ResNet18_unnorm/vanilla_orig_wBN_3407/adv_data/seed_3407/smallest_eps.csv  --unlearn_indices unlearn_indices/cifar/5000/seed_1.csv  --unlearn_method retrain

python hyperparam_tune.py --seed 3407 --method amun --dataset cifar --model ResNet18 --count 5000
```

**Comment by SunSwallow** [comment]
I also tried to reproduce the results by reimplementing AMUN in my code setting. and have similar questions. AMUN wants to shift the decision boundary buy jointly finetuning the model on D_F and D_A. However, how does the objective to guide the model to misclassify some samples to be forgotten and lower the UNLEARN ACC? Especially in full-class unlearning where the accuracy on D_F is always zero. 
 
I tried to add a hyperparameter beta to controls the weight of CE loss on D_F, and I found this works well with beta=0.4 for random subset unlearning, and beta=0, e.g., only finetuning on D_R and D_A, works for full-class unlearning. I fixed the training epoch to 5, used a cosine lr scheduler and adjusted the learning rate in my code setting.

**Comment by baigeiguai** [comment]
> I also tried to reproduce the results by reimplementing AMUN in my code setting. and have similar questions. AMUN wants to shift the decision boundary buy jointly finetuning the model on D_F and D_A. However, how does the objective to guide the model to misclassify some samples to be forgotten and lower the UNLEARN ACC? Especially in full-class unlearning where the accuracy on D_F is always zero.   I tried to add a hyperparameter beta to controls the weight of CE loss on D_F, and I found this works well with beta=0.4 for random subset unlearning, and beta=0, e.g., only finetuning on D_R and D_A, works for full-class unlearning. I fixed the training epoch to 5, used a cosine lr scheduler and adjusted the learning rate in my code setting.

Dear @SunSwallow:  

I am now confused about this method. During unlearning, the fine-tuning samples used by the model (including D_A, D_F, D_R) are all from the model f distribution, which means the following satisfies  before unlearning starts:
```
y_a =  f (x_a) , (x_a , y_a) \in D_A
y_f =  f (x_f) , (x_f , y_f) \in D_F
y_r =  f (x_r) , (x_r , y_r) \in D_R
```
So, I think the author's idea of fine-tuning model f with D_A,D_F,D_R to achieve the unlearn effect is unreasonable, the loss is close to 0 before fine-tuning even starts.  Given that I had limited knowledge about the unlearned domain prior to this, I would appreciate it if you could advise me on whether my concept is accurate. 

In addition, could you please kindly share some details and results that worked well after your reproduction ?

**Comment by SunSwallow** [comment]
@baigeiguai According to the paper, the idea of finetuning on D_F and D_A is motivated that the sample from D_F is similar to its adversarial sample from D_A in input space, but have different labels. So finetuning on both them may shift the decision boudary of original forgetting samples, or, lowering its prediction confidence as shown in Fig.3. 

I am also confused that why finetuning on D_A can realize unlearning. As the model is also finetuned on D_F, how does AMUN change the label of x_f? I think there are some key components we ignored, and I hope **the authors could clarify it**. 

In my implementation, I use a learning rate of 3e-4, adamw optimizer, cosine lr, and **the additional hyperparameter beta=0.4**, which can get a not bad results on CIFAR-10 using VGG16-BN. But i didn't read the codes carefully (the codes are not well organized.) and only implemented it according the paper and added the additional hyperparameter.
